flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestServerEndpointConfiguration.java

```diff
@@ -103,6 +126,9 @@ public static RestServerEndpointConfiguration fromConfiguration(Configuration co
 			}
 		}
 
-		return new RestServerEndpointConfiguration(address, port, sslEngine);
+		final Path uploadDir = Paths.get(config.getString(WebOptions.UPLOAD_DIR,
+			config.getString(WebOptions.TMP_DIR)));
```

Comment:
I think we should create a unique sub folder in order to avoid clashes between concurrent processes on the same machine. Will do this when merging this PR.